src/wazuh_modules/vulnerability_scanner/tests/databaseFeedManager/databaseFeedManager_test.cpp

```diff
@@ -0,0 +1,157 @@
+/*
+ * Wazuh databaseFeedManager
+ * Copyright (C) 2015, Wazuh Inc.
+ * September 22, 2023.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+#include "databaseFeedManager_test.h"
+#include "MockContentRegister.hpp"
+#include "MockIndexerConnector.hpp"
+#include "MockPolicyManager.hpp"
+#include "MockRouterSubscriber.hpp"
+#include "TrampolineContentRegister.hpp"
+#include "TrampolineIndexerConnector.hpp"
+#include "TrampolinePolicyManager.hpp"
+#include "TrampolineRouterSuscribe.hpp"
+#include "flatbuffers/flatbuffers.h"
+#include <string_view>
+
+using ::testing::Return;
+
+constexpr auto REMEDIATIONS_DB_DIR {"queue/vd"};
+
+std::shared_ptr<MockIndexerConnector> spIndexerConnectorMock;
+std::shared_ptr<MockPolicyManager> spPolicyManagerMock;
+std::shared_ptr<MockContentRegister> spContentRegisterMock;
+std::shared_ptr<MockRouterSubscriber> spRouterSubscriberMock;
+
+void DatabaseFeedManagerTest::SetUp()
+{
+    std::filesystem::create_directories(REMEDIATIONS_DB_DIR);
+};
+
+void DatabaseFeedManagerTest::TearDown()
+{
+    // Reset shared_ptr owner
+    spIndexerConnectorMock.reset();
+    spPolicyManagerMock.reset();
+    spContentRegisterMock.reset();
+    spRouterSubscriberMock.reset();
+    std::filesystem::remove_all(REMEDIATIONS_DB_DIR);
+};
+
+TEST_F(DatabaseFeedManagerTest, GetVulnerabilityRemediation_ValidData)
+{
+    // Test setup
+    const auto configurationParameters = R"( {"topicName": "topicNameTest"} )"_json;
+    std::shared_ptr<MockIndexerConnector> pIndexerConnectorMock;
+
+    spIndexerConnectorMock = std::make_shared<MockIndexerConnector>();
+
+    spPolicyManagerMock = std::make_shared<MockPolicyManager>();
+    EXPECT_CALL(*spPolicyManagerMock, getUpdaterConfiguration()).WillRepeatedly(Return(configurationParameters));
+
+    spContentRegisterMock = std::make_shared<MockContentRegister>(
+        configurationParameters.at("topicName").get<const std::string>(), configurationParameters);
+
+    spRouterSubscriberMock = std::make_shared<MockRouterSubscriber>(
+        configurationParameters.at("topicName").get<const std::string>(), "vulnerability_feed_manager");
+
```

Comment:
Using the following EXPECT_CALL you can remove the warning associated with this mock object.
`EXPECT_CALL(*spRouterSubscriberMock, subscribe(_));`

Reply:
Fixed. Thanks!